torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -297,6 +294,29 @@ std::shared_ptr<FutureMessage> ProcessGroupAgent::send(
     future->markCompleted();
   }
 
+  // Sending to ourselves: bypass the send logic and enqueue directly
+  // to our receving queue.
+  if (to.id_ == (worker_id_t)pg_->getRank()) {
+    TORCH_CHECK(!message.isShutdown(), "Shutting down self not supported");
+    threadPool_.run(std::bind(
+        [this](const Message& message) {
+          sendCounts_.increment(pg_->getRank());
+          // Unlike the other cases, need to add a tensor deleter, since the
+          // data outlives the scope of this function.
+          auto serializedPayload = new std::string(serialize(message));
+          enqueueRecv(RecvWork(
+              allWorkerInfo_[pg_->getRank()],
```

Comment:
there is a `RpcAgent::getWorkerInfo()` for this.

Reply:
I can change that. I was just copying the code a few lines below, later in this function. :)